### ISSUE TITLE
Missing Unlock before a branch of return err

### ISSUE THREAD

**Issue by lzhfromustc** [etcd-io/etcd#10863]
Hello all, I believe there is a missing Unlock bug in master branch. It is very easy to fix.
Location: https://github.com/etcd-io/etcd/blob/master/clientv3/watch.go Line: 387
```GO
func (w *watcher) RequestProgress(ctx context.Context) (err error) {
	ctxKey := streamKeyFromCtx(ctx)
	w.mu.Lock()
	if w.streams == nil {
		return fmt.Errorf("no stream found for context")
	}
```
Sorry I can't provide inputs or stack trace since I detected this bug by a static checker. 
I read related codes and it seems that the goroutine will exit after `RequestProgress` returns an error. However I think this might be a threat in the future, and after all, this bug is so easy to fix. XD

**Comment by jingyih** [comment]
@lzhfromustc Thanks! Do you feel like creating a pull request and help fix it?

**Comment by lzhfromustc** [comment]
Sure. I will request one. Thanks for letting me have the honor!

**Comment by jingyih** [comment]
For reference, fixed by #10866.
